shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm

```diff
@@ -41,18 +65,37 @@ - (void)addDelegate:(NSObject<FlutterPlugin>*)delegate {
   if (isPowerOfTwo([_pluginDelegates count])) {
     [_pluginDelegates compact];
   }
+
+  for (const auto& selector : lifeCycleSelectors) {
+    if ([delegate respondsToSelector:selector]) {
+      auto& p = _selectorTable[selector];
+      if (p.get() == nil) {
+        p.reset([[NSPointerArray weakObjectsPointerArray] retain]);
+      }
+
+      [p.get() addPointer:(__bridge void*)delegate];
```

Comment:
Note that I currently don't bother doing anything to compact these `NSPointerArray`s if plugins are destroyed.  I'm not familiar with plugin usage so don't know what to design for.  Are plugins expected to be created and destroyed a lot?  If so, then I think it'd be better to add a mechanism for plugins to unregister themselves.  Of course, if plugins are expected to be created and destroyed during the application's lifetime, then that's probably incompatible with this change's requirement of registering plugins during application startup.

Reply:
The lifespan of Flutter plugins should probably have been tied to that of the `FlutterViewController` with which they register and set up Dart/iOS communication through platform channels. In general, such Flutter views can be created and destroyed at will during the lifecycle of the app.

However, for Flutter-only apps created using `flutter create foo`, the life span of _the_ Flutter view is the same as that of the application---which is why we ended up in the current situation where plugins register once and have support for responding to all app delegate calls.

For add2app scenarios in particular, this coupling of the app lifecycle with the Flutter view lifecycle via plugins will have to be broken, I think. A first step was taken in #5173 which made it possible to delay plugin registration until the first Flutter view is created. That doesn't necessarily disqualify the present PR, but indicates that it is a step in a larger refactoring to fix an early design error.

Reply:
Hmm, ok.  Another possibility would be to add another mechanism to let clients apps specify which lifecycle events they want without having to implement handlers that forward to the `FlutterPluginAppLifeCycleDelegate` themselves.  OTOH, I'm a bit wary of adding multiple ways to do similar things.

And, of course, another possibility is to won't-fix #9984 and continue to live with the runtime warnings.

Reply:
@jamesderlin, as you know, I don't have a lot of familiarity with the specifics of iOS, but I definitely agree with the sentiment that plugins should exist within the lifespan of their associated Flutter view or views.  It may be worth grabbing a variety of plugin examples and discussing what impact that might have.  From an architecture and maintenance perspective, it may very well create issues that Flutter plugins need to be elevated to global app state when Flutter is being used, for example, as merely a list item in a feed.